src/schema/schema_alter.c

```diff
@@ -48,122 +56,6 @@ __alter_file(WT_SESSION_IMPL *session, const char *uri, const char *newcfg[])
 
 err:	__wt_free(session, config);
 	__wt_free(session, newconfig);
-	return (ret);
-}
-
-/*
- * __alter_colgroup --
- *	WT_SESSION::alter for a colgroup.
- */
-static int
-__alter_colgroup(
-    WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
-{
-	WT_COLGROUP *colgroup;
-	WT_DECL_RET;
-
-	WT_ASSERT(session, F_ISSET(session, WT_SESSION_LOCKED_TABLE));
-
-	/* If we can get the colgroup, perform any potential alterations. */
-	if ((ret = __wt_schema_get_colgroup(
-	    session, uri, false, NULL, &colgroup)) == 0)
-		WT_TRET(__wt_schema_alter(session, colgroup->source, cfg));
-
-	return (ret);
-}
-
-/*
- * __alter_index --
- *	WT_SESSION::alter for an index.
- */
-static int
-__alter_index(
-    WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
-{
-	WT_INDEX *idx;
-	WT_DECL_RET;
-
-	/* If we can get the index, perform any potential alterations. */
-	if ((ret = __wt_schema_get_index(
-	    session, uri, false, NULL, &idx)) == 0)
-		WT_TRET(__wt_schema_alter(session, idx->source, cfg));
-
-	return (ret);
-}
-
-/*
- * __alter_table --
- *	WT_SESSION::alter for a table.
- */
-static int
-__alter_table(WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
-{
-	WT_COLGROUP *colgroup;
-	WT_DECL_RET;
-	WT_TABLE *table;
-	const char *name;
-	u_int i;
-
-	name = uri;
-	(void)WT_PREFIX_SKIP(name, "table:");
-
-	WT_RET(__wt_schema_get_table(
-	    session, name, strlen(name), true, &table));
-
-	/*
-	 * Alter the column groups only if we are using the default
-	 * column group.  Otherwise the user should alter each
-	 * index or column group explicitly.
-	 */
-	if (table->ncolgroups == 0)
-		for (i = 0; i < WT_COLGROUPS(table); i++) {
```

Comment:
Similarly, `wt_schema_worker` does all this same work for us and gets us through each of the constituents of the table.